File: common/data/src/main/java/org/thingsboard/server/common/data/id/WidgetsBundleId.java
package org.thingsboard.server.common.data.id;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import org.thingsboard.server.common.data.EntityType;

public final class WidgetsBundleId extends UUIDBased implements EntityId {

    private static final long serialVersionUID = 1L;

    @JsonCreator
    public WidgetsBundleId(@JsonProperty("id") UUID id) {
        super(id);
    }

    @ApiModelProperty(position = 2, required = true, value = "string", example = "WIDGETS_BUNDLE", allowableValues = "WIDGETS_BUNDLE")
    @Override
    public EntityType getEntityType() {
        return EntityType.WIDGETS_BUNDLE;
    }
}

File: ui-ngx/src/app/shared/models/id/widgets-bundle-id.ts
import { EntityId } from './entity-id';
import { EntityType } from '@shared/models/entity-type.models';

export class WidgetsBundleId implements EntityId {
  entityType = EntityType.WIDGETS_BUNDLE;
  id: string;
  constructor(id: string) {
    this.id = id;
  }
}
